charts/sf-archival/charts/spark-manager/templates/deployment.yaml

```diff
@@ -26,7 +26,7 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
     {{- end }}
-    {{- if or (and (eq .Values.global.secrets.aws.enable true) (eq .Values.global.secrets.aws.use_iam_role true)) (and (eq .Values.global.secrets.gcs.enable true) (eq .Values.global.secrets.gcs.use_google_service_account true)) }}
+    {{- if or (and .Values.global.secrets.aws.enable .Values.global.secrets.aws.use_iam_role) .Values.global.secrets.gcs.enable }}
```

Comment:
service account condition needs to be added for azure too

Reply:
Service Account is to be only when IAM role or GKE service-account is used
Since Azure is replying on access keys which are being passed through cloud-env

I have allowed service-account to be created and used only when IAM way of auth is used

Reply:
Created the service account again since it is needed to create and delete executor/driver pods by spark-manager. Please check now